apps/desktop/electron/src/service/nuclia-cloud.ts

```diff
@@ -24,10 +26,15 @@ export class NucliaCloud {
     this.nuclia = new Nuclia(options);
   }
 
-  upload(originalId: string, filename: string, data: { buffer?: ArrayBuffer; metadata?: any }): Observable<boolean> {
-    if (data.buffer) {
-      const buffer = data.buffer;
-      const slug = sha256(originalId);
+  upload(
+    originalId: string,
+    filename: string,
+    data: { buffer?: ArrayBuffer; text?: TextField; metadata?: any },
```

Comment:
I don't understand: in `service/models.ts`, you added the following comment:
```
  // we cannot use the TextField from the SDK because we want to keep connectors independant
```
and here in `service/nuclia-cloud.ts` you're actually using `TextField` from the SDK…

Reply:
I meant the source connectors, not the destination ones, so the ones that external people might decide to implement themselves.